apps/api/src/domain/models/playlist/commands/add-audio-item-to-playlist/add-audio-item-to-playlist.command.integration.spec.ts

```diff
@@ -0,0 +1,235 @@
+                        [AggregateType.audioItem]: [existingAudioItem],
+                        [AggregateType.playlist]: [
+                            existingPlaylist.clone({
+                                items: [
+                                    {
+                                        resourceCompositeIdentifier:
+                                            existingAudioItem.getCompositeIdentifier(),
+                                    },
+                                ],
+                            }),
+                        ],
+                    }).fetchFullSnapshotInLegacyFormat(),
+                });
+            });
+        });
+
+        describe(`when the audio item does not exist`, () => {
+            it(`should fail with the expected errors`, async () => {
+                await assertCommandError(commandAssertionDependencies, {
+                    systemUserId: dummySystemUserId,
+                    buildCommandFSA: buildValidCommandFSA,
+                    initialState: new DeluxeInMemoryStore({
+                        [AggregateType.playlist]: [existingPlaylist],
+                        // no audio items!
+                    }).fetchFullSnapshotInLegacyFormat(),
+                    checkError: (error) => {
+                        assertErrorAsExpected(
+                            error,
+                            new CommandExecutionError([
+                                new InvalidExternalReferenceByAggregateError(
+                                    existingPlaylist.getCompositeIdentifier(),
+                                    [existingAudioItem.getCompositeIdentifier()]
+                                ),
+                            ])
+                        );
+                    },
+                });
+            });
+        });
+
+        describe(`when the playlist does not exist`, () => {
+            it(`should fail with the expected errors`, async () => {
+                await assertCommandError(commandAssertionDependencies, {
+                    systemUserId: dummySystemUserId,
+                    buildCommandFSA: buildValidCommandFSA,
+                    initialState: new DeluxeInMemoryStore({
+                        [AggregateType.audioItem]: [existingAudioItem],
+                        // no playlists!
+                    }).fetchFullSnapshotInLegacyFormat(),
+                    checkError: (error) => {
+                        assertErrorAsExpected(
+                            error,
+                            new CommandExecutionError([
+                                new AggregateNotFoundError(
+                                    existingPlaylist.getCompositeIdentifier()
+                                ),
+                            ])
+                        );
+                    },
+                });
+            });
+        });
+
+        describe(`when the aggregate composite identifier is of the wrong type`, () => {
```

Comment:
This is really important. I need to make sure other commands have the same test coverage. I might need to abstract this into a test helper.